src/ucp/wireup/wireup_ep.c

```diff
@@ -386,12 +512,12 @@ static UCS_CLASS_CLEANUP_FUNC(ucp_wireup_ep_t)
         uct_ep_destroy(self->sockaddr_ep);
     }
 
+    UCS_ASYNC_BLOCK(&worker->async);
     if (self->tmp_ep != NULL) {
         ucs_assert(!(self->tmp_ep->flags & UCP_EP_FLAG_USED));
-        ucp_ep_disconnected(self->tmp_ep, 1);
+        ucp_wireup_tmp_ep_destroy(ucp_ep, self, UCT_FLUSH_FLAG_CANCEL, NULL);
```

Comment:
why ret value is not checked here? request can leak here

Reply:
added a TODO that it will be replaced by `ucp_worker_discard_uct_ep()` when 5608 gets merged